pkg/providers/imagefamily/azlinux.go

```diff
@@ -83,6 +85,34 @@ func (u AzureLinux) DefaultImages() []types.DefaultImageOutput {
 	}
 }
 
+func (u AzureLinux) FIPSImages() []types.DefaultImageOutput {
+	// image provider will select these images in order, first match wins. This is why we chose to put Gen2 first in the defaultImages, as we prefer gen2 over gen1
+	return []types.DefaultImageOutput{
+		{
+			PublicGalleryURL:     AKSAzureLinuxPublicGalleryURL,
+			GalleryResourceGroup: AKSAzureLinuxResourceGroup,
+			GalleryName:          AKSAzureLinuxGalleryName,
+			ImageDefinition:      AzureLinux2Gen2FIPSImageDefinition,
+			Requirements: scheduling.NewRequirements(
+				scheduling.NewRequirement(v1.LabelArchStable, v1.NodeSelectorOpIn, karpv1.ArchitectureAmd64),
+				scheduling.NewRequirement(v1beta1.LabelSKUHyperVGeneration, v1.NodeSelectorOpIn, v1beta1.HyperVGenerationV2),
+			),
+			Distro: "aks-azurelinux-v2-gen2-fips",
```

Comment:
Double checked all the FIPS distros looks good:
[AgentBaker/pkg/agent/datamodel/types.go at 7d14215035b51860de586510afd396f19d7005be · Azure/AgentBaker](https://github.com/Azure/AgentBaker/blob/7d14215035b51860de586510afd396f19d7005be/pkg/agent/datamodel/types.go#L145C1-L199C81)